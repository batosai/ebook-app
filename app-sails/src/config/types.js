import { PropTypes as T } from 'react';

export const Library = T.shape({
  id: T.string.isRequired,
  name: T.string.isRequired,
});
