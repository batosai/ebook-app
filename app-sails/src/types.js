export const BOOKS_REQUEST = 'books:request';
export const BOOKS_SUCCESS = 'books:success';
export const BOOKS_FAILURE = 'books:failure';

export const BOOK_REQUEST = 'book:request';
export const BOOK_SUCCESS = 'book:success';
export const BOOK_EDIT_REQUEST = 'book:edit:request';
export const BOOK_EDIT_SUCCESS = 'book:edit:success';
export const BOOK_DELETE_REQUEST = 'book:delete:request';
export const BOOK_DELETE_SUCCESS = 'book:delete:success';

export const ASIDE_TOGGLE = 'aside:toggle';

export const COLLECTIONS_REQUEST = 'collections:request';
export const COLLECTIONS_SUCCESS = 'collections:success';
export const COLLECTIONS_FAILURE = 'collections:failure';
export const COLLECTIONS_FILTER = 'collections:filter';

export const COLLECTION_ADD_REQUEST = 'collection:add:request';
export const COLLECTION_EDIT_REQUEST = 'collection:edit:request';
export const COLLECTION_DELETE_REQUEST = 'collection:delete:request';

export const LIBRARIES_REQUEST = 'libraries:request';
export const LIBRARIES_SUCCESS = 'libraries:success';
export const LIBRARIES_FAILURE = 'libraries:failure';

export const LIBRARY_ADD_REQUEST = 'library:add:request';
export const LIBRARY_EDIT_REQUEST = 'library:edit:request';
export const LIBRARY_DELETE_REQUEST = 'library:delete:request';
