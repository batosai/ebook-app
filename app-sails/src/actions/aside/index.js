// import * as types from '../../types';
//
// export const toggleAsideAction = () => ({
//   type: types.ASIDE_TOGGLE
// });

import { createTypes, createActions } from '../../utils';

export const types = createTypes('ASIDE:TOGGLE');
export const actions = createActions(types);

export const dispatch = () => dispatch => {
  dispatch(actions.request());
};
