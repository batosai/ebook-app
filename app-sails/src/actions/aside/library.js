import { createTypes, createActions } from '../../utils';

export const types = createTypes('ASIDE:LIBRARY');
export const actions = createActions(types);

export const dispatch = id => dispatch => {
  dispatch(actions.request(id));
};
