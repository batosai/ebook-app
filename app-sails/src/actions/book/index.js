import * as create from './create';
import * as update from './update';
import * as remove from './remove';
import * as fetch from './fetch';
import * as upload from './upload';
import * as illustration from './illustration';

const dispatch = (dispatch, event) => {
  switch (event.verb) {
    case 'created':
      dispatch(create.actions.success(event.data));
      break;
    case 'updated':
      dispatch(update.actions.success(event.data));
      break;
    case 'destroyed':
      dispatch(remove.actions.success(event.id));
      break;
    default:
      console.warn(
        'Unrecognized socket event (`%s`) from server:',
        event.verb,
        event,
      );
  }
};

export { create, update, remove, fetch, dispatch, illustration, upload };
