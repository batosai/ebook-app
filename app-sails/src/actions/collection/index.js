import * as create from './create';
import * as update from './update';
import * as remove from './remove';
import * as fetchAll from './fetchAll';
export { create, update, remove, fetchAll };
