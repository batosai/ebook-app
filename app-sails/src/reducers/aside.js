import { aside } from '../actions';

const initialState = {
  open: false,
};

const reducers = (state=initialState, {type, payload}) => {
  switch (type) {
    case aside.types.REQUEST:
      return Object.assign({}, state, {open: !state.open});
    default:
      return state;
  }
}

export default reducers;
