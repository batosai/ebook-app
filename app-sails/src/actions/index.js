// import { toggleAsideAction } from './aside';
// export { toggleAsideAction };

// import { booksRequestAction } from './books';
// export { booksRequestAction };
//
// import { bookRequestAction, bookEditRequestAction, bookDeleteRequestAction } from './book';
// export { bookRequestAction, bookEditRequestAction, bookDeleteRequestAction };
//
// import { collectionsRequestAction, collectionByLibraryAction, collectionAddRequestAction, collectionEditRequestAction, collectionDeleteRequestAction } from './collections';
// export { collectionsRequestAction, collectionByLibraryAction, collectionAddRequestAction, collectionEditRequestAction, collectionDeleteRequestAction };
//
// import { librariesRequestAction, libraryAddRequestAction, libraryEditRequestAction, libraryDeleteRequestAction } from './libraries';
// export { librariesRequestAction, libraryAddRequestAction, libraryEditRequestAction, libraryDeleteRequestAction };


import * as aside from './aside';
import * as book from './book';
import * as collection from './collection';
import * as library from './library';
export {
  aside,
  book,
  collection,
  library,
}
