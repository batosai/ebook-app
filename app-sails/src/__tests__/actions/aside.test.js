import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import * as actions from '../../actions/aside';
import * as types from '../../types';

const middlewares = [ thunk ];
const mockStore = configureMockStore(middlewares);

describe('Action', () => {
  it('to toggle aside', () => {

    const store = mockStore({
      aside: {
        open: false
      }
    });

    const expectedAction = {
      type: types.ASIDE_TOGGLE
    };

    expect(actions.toggleAsideAction()).toEqual(expectedAction);
    expect(actions.toggleAsideAction()).toMatchSnapshot();


    // return store.dispatch(actions.asideToggle())
    //   .then(() => { // return of async actions
    //     expect(store.getActions()).toEqual(expectedActions)
    //   })
  });
});

// TODO TESTS A FAIRE
// actions -> ok
// reducers -> ok
// components
// containers ?
// dispatchers ?
// routes ?
// middleware ? (socket.io)

// http://redux.js.org/docs/basics/ExampleTodoList.html
// http://redux.js.org/docs/recipes/WritingTests.html
// https://facebook.github.io/jest/docs/getting-started.html
// https://facebook.github.io/jest/docs/tutorial-react.html#content
