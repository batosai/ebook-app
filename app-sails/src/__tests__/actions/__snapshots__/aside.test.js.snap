exports[`Action to toggle aside 1`] = `
Object {
  "type": "aside:toggle",
}
`;
